plenum/client/client.py

```diff
@@ -603,6 +602,34 @@ def hasSufficientConnections(self):
     def hasAnyConnections(self):
         return len(self.nodestack.conns) > 0
 
+    def can_send_write_requests(self):
+        if not Mode.is_done_discovering(self.mode):
+            return False
+        if not self.hasSufficientConnections:
+            return False
+        return True
+
+    def can_send_read_requests(self):
+        if not Mode.is_done_discovering(self.mode):
+            return False
+        if not self.hasAnyConnections:
+            return False
+        return True
+
+    def can_send_request(self, request):
```

Comment:
Can't it be implemented using  can_send_write_requests and can_send_read_requests?

Reply:
It has some additional logic not covered by send read and send write. So, can_send_read and can_send_write can be used, although not straightforward, so I would prefer to let it be as it is.